arangod/Transaction/Methods.cpp

```diff
@@ -666,6 +668,36 @@ std::pair<bool, bool> transaction::Methods::findIndexHandleForAndNode(
       bestSupportsFilter = supportsFilter;
       bestSupportsSort = supportsSort;
     }
+  };
+
+  if (hint.type() == aql::IndexHint::HintType::Simple) {
+    std::vector<std::string> const& hintedIndices = hint.hint();
+    for (std::string const& hinted : hintedIndices) {
+      std::shared_ptr<Index> matched;
+      for (std::shared_ptr<Index> const& idx : indexes) {
+        if (idx->name() == hinted) {
+          matched = idx;
+          break;
+        }
+      }
+
+      if (matched != nullptr) {
+        considerIndex(matched);
+        if (bestIndex != nullptr) {
+          break;
+        }
+      }
+    }
+
+    if (hint.isForced() && bestIndex == nullptr) {
+      THROW_ARANGO_EXCEPTION(TRI_ERROR_QUERY_FORCED_INDEX_HINT_UNUSABLE);
```

Comment:
Ideally we could also give a description of the non-found index hint here, e.g. by putting the supplied IndexHint into a human-readable form and putting it into the error message.